kaldb/src/main/java/com/slack/kaldb/chunkManager/ChunkManagerBase.java

```diff
@@ -45,10 +54,14 @@
    * Revisit the thread pool settings if this becomes a perf issue. Also, we may need
    * different thread pools for indexer and cache nodes in the future.
    */
-  private static ExecutorService queryThreadPool() {
-    return Executors.newFixedThreadPool(
-        Runtime.getRuntime().availableProcessors(),
-        new ThreadFactoryBuilder().setNameFormat("chunk-manager-query-%d").build());
+  private static ListeningExecutorService queryThreadPool() {
+    // todo - consider making the thread count a config option; this would allow for more
+    //  fine-grained tuning, but we might not need to expose this to the user if we can set sensible
+    //  defaults
+    return MoreExecutors.listeningDecorator(
+        Executors.newFixedThreadPool(
+            Math.max(1, Runtime.getRuntime().availableProcessors() - 2),
+            new ThreadFactoryBuilder().setNameFormat("chunk-manager-query-%d").build()));
```

Comment:
Similar to the previous comment, I [just added](https://github.com/slackhq/kaldb/commit/8bf26d9e0034565331584340637ac7226949d05f#diff-e1190a6a806d660135b17a398bb14ba38c3554852e456cfa1812bc836e87ae7cR52) an uncaught exception handler here that we should retain.

Reply:
Done.